storage/imxrt-flash/flashsrv.c

```diff
@@ -166,6 +168,62 @@ static int flashsrv_chipErase(uint8_t fID)
 }
 
 
+static uint32_t flashsrv_crc32Update(uint32_t crc32, const uint8_t *data, size_t len)
+{
+	unsigned int i;
+
+	while (len--) {
+		crc32 ^= *data++;
+		for (i = 0; i < 8; i++) {
+			crc32 = (crc32 >> 1) ^ ((crc32 & 1) ? 0xedb88320 : 0);
+		}
+	}
+
+	return crc32;
+}
+
+
+static int flashsrv_calcCrc32(uint8_t fID, size_t offset, size_t len, uint32_t *crc32)
+{
+	flashsrv_memory_t *mem;
+	uint8_t *buf;
+	uint32_t tmp;
+	int res;
+
+	res = flashsrv_getFlashMemory(fID, &mem);
+	if (res < 0) {
+		return res;
+	}
+
+	buf = malloc(CRC32_BUFSZ);
```

Comment:
The imxrt-flash is a legally relevant part of fw. All allocated memory should be known at initialization stage or compile time. Using malloc/free at each call to function may lead to memory fragmentation during long-term operation (MISRA Dir 4.12 req.).
Btw. NOR page size is 256 bytes @ 4KiB sector size, you can use smaller buffer for it e.g. of page size 256 bytes if allocated statically to save ram space.